dogma.go

```diff
@@ -79,46 +83,60 @@ type service struct {
 	client *Client
 }
 
-// NewClient returns a Central Dogma client with the specified baseURL, username and password.
-func NewClient(baseURL, username, password string) (*Client, error) {
+// NewClientWithToken returns a Central Dogma client which communicates the server at baseURL, using the specified
+// token and transport. If transport is nil, http2.Transport is used by default.
+func NewClientWithToken(baseURL, token string, transport http.RoundTripper) (*Client, error) {
 	normalizedURL, err := normalizeURL(baseURL)
 	if err != nil {
 		return nil, err
 	}
 
+	if len(token) == 0 {
+		return nil, errors.New("token should not be empty")
+	}
+
 	config := oauth2.Config{Endpoint: oauth2.Endpoint{TokenURL: normalizedURL.String() + pathLogin}}
-	token, err := config.PasswordCredentialsToken(context.Background(), username, password)
+	tokenSource := config.TokenSource(context.Background(), &oauth2.Token{AccessToken: token})
+	client, err := newOauth2HTTP2Client(normalizedURL.String(), tokenSource, transport)
 	if err != nil {
 		return nil, err
 	}
 
-	return newClientWithHTTPClient(normalizedURL.String(), config.Client(context.Background(), token))
+	return newClientWithHTTPClient(normalizedURL, client)
 }
 
-// NewClientWithToken returns a Central Dogma client with the specified baseURL and token.
-func NewClientWithToken(baseURL, token string) (*Client, error) {
-	normalizedURL, err := normalizeURL(baseURL)
-	if err != nil {
-		return nil, err
+func newOauth2HTTP2Client(normalizedURL string,
+	tokenSource oauth2.TokenSource, transport http.RoundTripper) (*http.Client, error) {
+	if transport == nil {
+		transport = defaultHTTP2Transport(normalizedURL)
 	}
 
-	config := oauth2.Config{Endpoint: oauth2.Endpoint{TokenURL: normalizedURL.String() + pathLogin}}
-	oauthToken := &oauth2.Token{AccessToken: token}
-
-	return newClientWithHTTPClient(normalizedURL.String(), config.Client(context.Background(), oauthToken))
+	return &http.Client{
+		Transport: &oauth2.Transport{
+			Base:   transport,
+			Source: oauth2.ReuseTokenSource(nil, tokenSource),
+		},
+	}, nil
+}
+
+func defaultHTTP2Transport(normalizedURL string) *http2.Transport {
```

Comment:
We could make this function public for user reusing

Reply:
Fixed Thank you~!